src/ray/common/id.cc

```diff
@@ -194,6 +194,14 @@ TaskID TaskID::ForNormalTask(const JobID &job_id, const TaskID &parent_task_id,
   return TaskID::FromBinary(data);
 }
 
+TaskID TaskID::ForExecution(const TaskID &task_id, int64_t execution_counter) {
+  std::string data_str;
+  std::copy_n(task_id.Data(), TaskID::kLength, std::back_inserter(data_str));
+  auto data = reinterpret_cast<uint8_t *>(data_str.data());
+  data[TaskID::kLength - 1] += execution_counter;
```

Comment:
How about we reinterpret_cast<uint64_t> instead to avoid this?

Btw, it would be great to initialize the last byte of task_id to zero by default, so that the counter is more readable.